app/shared/event/event_bus.py

```diff
@@ -0,0 +1,29 @@
+class EventBus():
+    def __init__(self):
+        self._subscribers = {}
+
+    async def publish(self, event):
+        event_classes = [x for x in list(
+            map(lambda x: x.__name__, type(event).mro())) if x != 'object']
+
+        subscribers = []
+        for event_class in event_classes:
+            if not event_class in self._subscribers.keys():
```

Comment:
`not x in y` is not the "right" way.
`x not in y` is better.
and a key not in keys can be checked without `.keys()` if _subscribers is a dict:
```suggestion
            if event_class not in self._subscribers:
```